Further cleaned locations based on Charles update

utkuozbulak opened the pull request.

utkuozbulak committed:
Further cleaned locations based on Charles update

utkuozbulak commented on src/cleaning_functions.py:
```diff
@@ -8,13 +10,16 @@ def get_bin(x, n):
 def get_one_hot_encoded_feature(feature):
-    feature_cleaned = pandas_vector_to_list(feature)
-    feature_set = set(feature_cleaned)  # Get unique values
+    if type(feature) == "<class 'list'>":  # Input is list
```
To support list inputs with pandas dataframe

charlienewey replied:
if isinstance(feature, list)

charlienewey commented on src/cleaning_functions.py:
```diff
@@ -24,7 +29,10 @@ def get_one_hot_encoded_feature(feature):
 
 def get_binary_encoded_feature(feature):
-    feature_as_list = pandas_vector_to_list(feature)
+    if str(type(feature)) == "<class 'list'>":  # Input is list
```
`if isinstance(feature, list)`

utkuozbulak committed:
Pep update, obviously

charlienewey reviewed (commented):
bad practice to do `type(x) == "<class ...>"

utkuozbulak committed:
Updated type comparison with isinstance

utkuozbulak committed:
Few updates on code

utkuozbulak commented:
@charlienewey ' bad practice to do `type(x) == "<class ...>" ' 
Updated :heart:

utkuozbulak commented:
This is a basic cleaning on stuff google couldn't find but was obvious. Some stats are below, uncomment print statements to see what is updated

Some stats:

field 'town'
total empty records: 148409
updated: 77348

Examples:
Found on raw location, updated on cleaned
Found: Bristol  Updated:  Bristol
Found: Grimsby  Updated:  Grimsby
Found: Edinburgh & Lothian  Updated:  Edinburgh
Found: City of London  Updated:  London

field: 'region'
total empty records: 71815
updated: 30280

Examples:
Found on raw location, updated on cleaned:
Found: Hampshire-wide  Updated:  Hampshire
Found: Staffordshire Staffordshire England  Updated:  Staffordshire
Found: Essex  Updated:  Essex
Found: Berkshire  Updated:  Berkshire
Found: Basingstoke Hampshire England  Updated:  Hampshire

charlienewey committed:
Merge branch 'master' into location_cleaning
